src/wms/src/main/java/org/geoserver/wms/WMS.java

```diff
@@ -1835,16 +1843,42 @@ private void validateElevations(
 
         if (elevations != null) {
             DimensionFilterBuilder builder = new DimensionFilterBuilder(ff);
-            List<Object> explicitEleevations =
-                    elevations.stream().filter(v -> v != null).collect(Collectors.toList());
-            getElevationFilter(explicitEleevations, typeInfo, builder);
-            Filter filter = builder.getFilter();
-            if (DataUtilities.first(fs.getFeatures(filter)) == null) {
+            List<Object> explicitElevations =
+                    elevations.stream().filter(Objects::nonNull).collect(Collectors.toList());
+            Query dimensionQuery =
+                    getDimensionQuery(
+                            getElevationFilter(explicitElevations, typeInfo, builder),
+                            elevationInfo,
+                            typeInfo.getFeatureType().getName().getLocalPart());
+            if (DataUtilities.first(fs.getFeatures(dimensionQuery)) == null) {
                 throwInvalidDimensionValue(request, ELEVATION, ELEVATION);
             }
         }
     }
 
+    /**
+     * wraps the given filter in a query for dimension requests.
```

Comment:
```suggestion
     * Wraps the given filter in a query for dimension requests.
```

Javadoc comments should start with a capital (it's a regular sentence). Can you please also fix the test method javadocs?

Reply:
Done.